protocols/kad/src/kbucket.rs

```diff
@@ -120,6 +121,26 @@ impl KBucketsPeerId<PeerId> for Multihash {
     }
 }
 
+impl KBucketsPeerId for KadHash {
+    fn distance_with(&self, other: &Self) -> u32 {
+        <Multihash as KBucketsPeerId<Multihash>>::distance_with(self.hash(), other.hash())
+    }
+
+    fn max_distance() -> NonZeroUsize {
+        <Multihash as KBucketsPeerId>::max_distance()
+    }
+}
+
+impl KBucketsPeerId<KadHash> for Multihash {
+    fn distance_with(&self, other: &KadHash) -> u32 {
+        <Multihash as KBucketsPeerId<Multihash>>::distance_with(self, other.hash())
+    }
+
+    fn max_distance() -> NonZeroUsize {
+        <Multihash as KBucketsPeerId>::max_distance()
```

Comment:
Reading your PR makes me realize that it doesn't make sense to have `max_distance()` when you implement `KBucketsPeerId<A>` for `B` and `A != B`. I guess this could be improved, but that's out of scope of this PR.

Reply:
I guess you could take the largest of the two and calculate distance there, see my comment above.